packages/@react-spectrum/combobox/test/ComboBox.test.js

```diff
@@ -0,0 +1,1486 @@
+      });
+
+      it('has a uncontrolled open state via defaultOpen', function () {
+        let {getByRole} = renderComboBox({defaultOpen: true});
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        expect(() => getByRole('listbox')).toThrow();
+
+        act(() => {
+          combobox.focus();
+          jest.runAllTimers();
+        });
+
+        let listbox = getByRole('listbox');
+        expect(onOpenChange).toBeCalledTimes(0);
+        testComboBoxOpen(combobox, button, listbox, 0);
+
+        act(() => {
+          triggerPress(button);
+          jest.runAllTimers();
+        });
+
+        expect(() => getByRole('listbox')).toThrow();
+        expect(onOpenChange).toBeCalledTimes(1);
+        expect(onOpenChange).toHaveBeenCalledWith(false);
+        expect(button).toHaveAttribute('aria-expanded', 'false');
+        expect(button).not.toHaveAttribute('aria-controls');
+        expect(combobox).not.toHaveAttribute('aria-controls');
+      });
+    });
+
+    describe('button click', function () {
+      it('keeps focus within the textfield after opening the menu', function () {
+        let {getByRole} = renderComboBox();
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        expect(() => getByRole('listbox')).toThrow();
+
+        act(() => {
+          combobox.focus();
+          triggerPress(button);
+          jest.runAllTimers();
+        });
+
+        let listbox = getByRole('listbox');
+        expect(listbox).toBeTruthy;
+        expect(document.activeElement).toBe(combobox);
+      });
+
+      // Double check this test
+      // it('fires onFilter if there are no items loaded yet', function () {
+      //   let {getByRole} = render(
+      //     <Provider theme={theme}>
+      //       <ComboBox label="Test" onOpenChange={onOpenChange} onFilter={onFilter} items={[]}>
+      //         {(item) => <Item key={item.key}>{item.name}</Item>}
+      //       </ComboBox>
+      //     </Provider>
+      //   );
+
+      //   let button = getByRole('button');
+      //   act(() => {
+      //     triggerPress(button);
+      //     jest.runAllTimers();
+      //   });
+
+      //   expect(onFilter).toHaveBeenCalled();
+
+      //   expect(onOpenChange).toHaveBeenCalledWith(true);
+      //   expect(onOpenChange).toHaveBeenCalledTimes(1);
+
+      //   expect(() => getByRole('listbox')).toThrow();
+      // });
+
+      it('focuses first item if there are items loaded', function () {
+        let {getByRole} = renderComboBox();
+
+        let button = getByRole('button');
+        let combobox = getByRole('combobox');
+        act(() => {
+          triggerPress(button);
+          jest.runAllTimers();
+        });
+
+        expect(onOpenChange).toHaveBeenCalledWith(true);
+
+        let listbox = getByRole('listbox');
+        testComboBoxOpen(combobox, button, listbox, 0);
+      });
```

Comment:
I don't see a check for which items was focused.

Reply:
it gets checked in the testComboBoxOpen function (the 0 meaning the first item in the list and the check is the aria-activedescendant assert)